### PR TITLE
[86dvfm529] Tasks Backend

### PR DESCRIPTION
**Pull request by tomisueldo**
# ⚡ Tasks Backend - [86dvfm529](https://app.clickup.com/t/86dvfm529) ⚡

## 💻 What type of change is this?

- [x] 💎 Feature

## ⭐ Description

Form requests, controllers, and actions for the Tasks resource

## 📷 Screenshots

image

image

image

image

### To review
- [ ] I have tested this change locally in multiple screen sizes
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] If my task include an endpoint, I add the endpoint to Hopscotch/Postman Project
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: routes/api.php [hunk 1]
```diff
@@ -44,5 +48,11 @@
         Route::post('/', StoreEmployeeController::class)->name('employees');
     });
 
-Route::post('/tasks', function () {
-})->name('tasks');
+Route::prefix('tasks')
+    ->middleware([])
```
Comment (mateorodriguezcastillo) [inline routes/api.php]: Remove if using empty array

File: src/Backoffice/Task/App/Controllers/GetTaskController.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+use Illuminate\Http\JsonResponse;
+use Lightit\Backoffice\Task\App\Transformers\TaskTransformer;
+use Lightit\Backoffice\Task\Domain\Actions\GetTaskAction;
+use Lightit\Backoffice\Task\Domain\Models\Task;
+
+class GetTaskController
+{
+    public function __invoke(Task $task, GetTaskAction $getTaskAction): JsonResponse
+    {
+        $getTaskAction->execute();
```
Comment (mateorodriguezcastillo) [inline src/Backoffice/Task/App/Controllers/GetTaskController.php]: You can omit this action if you're returning directly $task

File: src/Backoffice/Task/App/Controllers/ListTaskController.php [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+use Illuminate\Http\JsonResponse;
+use Lightit\Backoffice\Task\App\Transformers\TaskTransformer;
+use Lightit\Backoffice\Task\Domain\Actions\ListTaskAction;
+
+class ListTaskController
+{
+    public function __invoke(ListTaskAction $listTaskAction): JsonResponse
+    {
+        $employees = $listTaskAction->execute();
```
Comment (mateorodriguezcastillo) [inline src/Backoffice/Task/App/Controllers/ListTaskController.php]: Rename variable

File: src/Backoffice/Task/App/Controllers/StoreTaskController.php [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+use Illuminate\Http\JsonResponse;
+use Lightit\Backoffice\Task\App\Request\StoreTaskRequest;
+use Lightit\Backoffice\Task\App\Transformers\TaskTransformer;
+use Lightit\Backoffice\Task\Domain\Actions\StoreTaskAction;
+
+class StoreTaskController
+{
+    public function __invoke(StoreTaskRequest $request, StoreTaskAction $storeTaskAction): JsonResponse
+    {
+        $employee = $storeTaskAction->execute($request->toDto());
```
Comment (mateorodriguezcastillo) [inline src/Backoffice/Task/App/Controllers/StoreTaskController.php]: Idem rename

File: src/Backoffice/Task/App/Controllers/UpdateTaskController.php [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+use Lightit\Backoffice\Task\App\Request\UpdateTaskRequest;
+use Lightit\Backoffice\Task\App\Transformers\TaskTransformer;
+use Lightit\Backoffice\Task\Domain\Actions\UpdateTaskAction;
+use Lightit\Backoffice\Task\Domain\Models\Task;
+
+class UpdateTaskController
+{
+    public function __invoke(Task $task, UpdateTaskRequest $request, UpdateTaskAction $storeTaskAction): JsonResponse
+    {
+        $employee = $storeTaskAction->execute($task, $request->toDto());
```
Comment (mateorodriguezcastillo) [inline src/Backoffice/Task/App/Controllers/UpdateTaskController.php]: Idem

File: src/Backoffice/Task/App/Request/UpdateTaskRequest.php [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    public const EMPLOYEE_ID = 'employeeId';
+
+    public function rules(): array
+    {
+        return [
+            self::TITLE => ['required', 'string'],
+            self::DESCRIPTION => ['required', 'string'],
+            self::STATUS => ['required', new Enum(TaskStatus::class)],
+            self::EMPLOYEE_ID => ['required', 'numeric', 'exists:employees,id'],
+
```
Comment (Tincho44) [inline src/Backoffice/Task/App/Request/UpdateTaskRequest.php:30]: empty space

**Review by mateorodriguezcastillo — APPROVED** [review]

File: routes/api.php (hunk 1 above)
Comment (Tincho44) [inline routes/api.php:52]: why empty middleware?

**Pull request merged**